messaging-core/src/main/java/org/fenixedu/messaging/core/domain/Message.java

```diff
@@ -191,95 +192,82 @@ public MessageBuilder content(LocalizedString subject, LocalizedString textBody,
             return this;
         }
 
-        private static <T> void filteredAdd(Stream<T> items, Collection<T> collection) {
-            items.filter(Objects::nonNull).forEach(collection::add);
+        private static <T> void filteredAdd(Stream<T> stream, Predicate<T> filter, Set<T> set) {
+            if(stream != null) {
```

Comment:
Ok, but that implies adding null checks to 4 other methods that would otherwise simply call this one. And that would also be weird since it would differ from the other 2 similar methods for Collections and Arrays. Do I do it anyway?
One more thing, I overlooked that I'm using the same method types in the sender builder. I'm thinking of moving these private methods to MessagingSystem.Util as protected and use them there as well. Is it ok? I'm thinking of squashing and pushing that to this branch.